lib/absinthe/middleware/dataloader.ex

```diff
@@ -3,15 +3,46 @@ if Code.ensure_loaded?(Dataloader) do
     @behaviour Absinthe.Middleware
     @behaviour Absinthe.Plugin
 
+    @dataloader_start_event [:absinthe, :dataloader, :resolve, :start]
+    @dataloader_stop_event [:absinthe, :dataloader, :resolve, :stop]
+
+    @impl Absinthe.Plugin
     def before_resolution(%{context: context} = exec) do
       context =
         with %{loader: loader} <- context do
-          %{context | loader: Dataloader.run(loader)}
+          start_time = System.monotonic_time()
+
+          emit_start_event(start_time, exec, loader)
+          updated_loader = Dataloader.run(loader)
+          emit_stop_event(start_time, exec, loader)
```

Comment:
Specifically, I think we should move these start and end events into https://github.com/absinthe-graphql/absinthe/blob/master/lib/absinthe/phase/document/execution/resolution.ex#L78 and it would be something like:

```
  defp run_callbacks(plugins, callback, acc, true) do
    Enum.reduce(plugins, acc, fn plugin, acc ->
      start_time = System.monotonic_time()
              :telemetry.execute(
          [:absinthe, :plugin, callback, :start]
          %{start_time: start_time},
          %{plugin: plugin, acc: acc}
        )
      acc = apply(plugin, callback, [acc])
      # stop event
     acc
    end)
  end
```